File: add/codesnippet/VisualBasic/p-system.web.services.de_32_1.vb
         Dim myService As Service
         Dim myPortCollection As PortCollection

         Dim myServiceDescription As ServiceDescription = _
            ServiceDescription.Read("MathServiceItem_vb.wsdl")

         Console.WriteLine("Total number of services : " & _
            myServiceDescription.Services.Count.ToString)

         Dim i As Integer
         For i = 0 to myServiceDescription.Services.Count - 1
            myService = myServiceDescription.Services(i)
            Console.WriteLine("Name : " & myService.Name)

            myPortCollection = myService.Ports

            ' Create an array of ports.
            Console.WriteLine(ControlChars.NewLine & "Port collection :")
            Dim i1 As Integer
            For i1 = 0 to myService.Ports.Count - 1
               Console.WriteLine("Port[" & i1.ToString & "] : " & _
                  myPortCollection(i1).Name)
            Next

File: add/codesnippet/CPP/p-system.web.services.de_32_1.cpp
      Service^ myService;
      PortCollection^ myPortCollection;
      ServiceDescription^ myServiceDescription = ServiceDescription::Read( "MathServiceItem_cs.wsdl" );
      Console::WriteLine( "Total number of services : {0}", myServiceDescription->Services->Count );
      for ( int i = 0; i < myServiceDescription->Services->Count; ++i )
      {
         myService = myServiceDescription->Services[ i ];
         Console::WriteLine( "Name : {0}", myService->Name );
         myPortCollection = myService->Ports;

         // Create an array of ports.
         Console::WriteLine( "\nPort collection :" );
         for ( int i1 = 0; i1 < myService->Ports->Count; ++i1 )
         {
            Console::WriteLine( "Port[{0}] : {1}", i1, myPortCollection[ i1 ]->Name );
         }